tests/test_project_access.py

```diff
@@ -405,3 +429,62 @@ def test_fix_access_unitadmin_valid_email_unituser(client):
         project_id=project.id, user_id="unituser"
     ).first()
     assert user_project_key_row
+
+
+def test_fix_access_unitadmin_valid_email_unituser_no_project(client):
+    """Unit Admin giving access to unituser - ok. No project."""
+    # Remove ProjectUserKeys row for specific project and user
+    project: models.Project = delete_project_user(
+        project_id="public_project_id", user_id="unituser"
+    )
+
+    # Fix access for user with no project specified
+    token = tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(client)
+    response = client.post(
+        tests.DDSEndpoint.PROJECT_ACCESS,
+        headers=token,
+        json={"email": "unituser1@mailtrap.io"},
+    )
+    assert response.status_code == http.HTTPStatus.OK
+
+    # Verify that the projectuserkey row is fixed
+    user_project_key_row = models.ProjectUserKeys.query.filter_by(
+        project_id=project.id, user_id="unituser"
+    ).first()
+    assert user_project_key_row
+
+
+def test_fix_access_unitadmin_valid_email_researcher_no_projectuser_row(client):
+    """Unit Admin giving access to researcher where there is no row in ProjectUsers table."""
+    # Remove ProjectUserKeys row for specific project and user
+    project: models.Project = delete_project_user(
+        project_id="public_project_id", user_id="researchuser"
+    )
+
+    # Delete projectuser row and verify that it's deleted for user
+    project_users_row = models.ProjectUsers.query.filter_by(
```

Comment:
This part of 
 _Delete projectuser row and verify that it's deleted for user_

can be removed as it is encapsulated by the helper function :)

Reply:
The helper function uses the `ProjectUserKeys` table while this uses the `ProjectUsers` table. 
We could techically add add another argument to the function to specify which table, I'll have a look and see if it simplifies.

Reply:
ahhh I never saw that difference, sorry. Forget what I said.

Reply:
No worries, very similar!

Reply:
I added the variable to the helper function now though, I'll tag you as a reviewer again